src/python/pants/option/global_options.py

```diff
@@ -836,6 +839,13 @@ def register_bootstrap_options(cls, register):
             advanced=True,
             help="Enable the streaming remote execution client (experimental).",
         )
+        register(
+            "--remote-execution-overall-deadline-secs",
+            type=int,
+            default=DEFAULT_EXECUTION_OPTIONS.remote_execution_overall_deadline_secs,
+            advanced=True,
+            help="Overall timeout in seconds for each remote execution request from time of submission",
```

Comment:
Similar to the existing `queue_buffer_time`, this should probably be time added to the process's own timeout... and consequently, it should probably have a much lower default.

But, also similar to the existing queue buffer time, it's possible that this doesn't need to be end-user configurable?

Reply:
This is intended as a "things are stuck" timeout, this it runs from when the request is first given to StreamingCommandRunner whereas the per-process timeout seems like it is "once it starts executing, enforce this deadline on its actual execution." Thus are two similar, but different concepts.

I'd prefer to retain this timeout and leave it configurable since I don't know what a reasonable choice would be yet. Indeed, in local testing with the toolchain repo, I hit the 10 minute limit when using a local BuildBarn so would need even in this case to modify the timeout.

Reply:
Ah but I see your point, it should be per-process timeout plus the deadline timeout so the timeout triggers if stuck for X minutes past the per-process timeout.

Reply:
I need it at least 10 minutes for my local testing (but the RE server in question is resource constrained running in a VM on my laptop). Thoughts on what the default for Pants should be?

Reply:
Oh, sorry... I misread this as one hour. We used to have a default per-process timeout of 900 seconds, but we removed it because of how frustrating it was for a long running process to time out.

That's relevant here: we might want to consider not applying this overall timeout if the process itself doesn't have a timeout. Because if a process doesn't have a timeout configured, then any value is potentially problematic?

So... how about:
1. if a process has a timeout configured, we add this value (defaulting to 600 seconds would be fine there) and apply it as an overall timeout.
2. if a process has no timeout configured, do not apply an overall timeout either.

Reply:
Since there is always the potential of the server becoming non-responsive, no timeout ever when the process does not express one does not seem like a great choice either. What if we leave the choice more up to the user? So:

1. If the process has a timeout configured, use that plus any configured deadline timeout.

2. if a process has no timeout configured but the user has enabled deadline timeouts explicitly, then use the configured deadline timeout.

3. Otherwise, no timeout.

And your change to pass any per-process timeout to the server should also be landed.

Reply:
Seems reasonable.

Reply:
> I'd also be fine with just setting a very high default (60 minutes?) and always applying it.

That is probably simpler in practice. And if the user wants to lower it, they can configure it lower. Will try with 60 minutes as the default.